engines/advancedDetector.h

```diff
@@ -291,45 +379,49 @@ class AdvancedMetaEngineDetection : public MetaEngineDetection {
-	 * @param language	restrict results to specified language
-	 * @param platform	restrict results to specified platform
-	 * @param extra		restrict results to specified extra string (only if kADFlagUseExtraAsHint is set)
-	 * @return	list of ADGameDescription pointers corresponding to matched games
+	 * @param parent    Parent node of this file node.
+	 * @param allFiles  List of all present files, as computed by the @ref composeFileHashMap.
+	 * @param language  Restrict results to the specified language.
+	 * @param platform  Restrict results to the specified platform.
+	 * @param extra     Restrict results to the specified @c extra string (only if @ref kADFlagUseExtraAsHint is set).
+	 *
+	 * @return A list of @ref ADGameDescription pointers corresponding to the matched games.
 	 */
 	virtual ADDetectedGames detectGame(const Common::FSNode &parent, const FileMap &allFiles, Common::Language language, Common::Platform platform, const Common::String &extra) const;
 
 	/**
-	 * Iterates over all ADFileBasedFallback records inside fileBasedFallback.
-	 * This then returns the record (or rather, the ADGameDescription
+	 * Iterate over all @ref ADFileBasedFallback records inside @p fileBasedFallback
+	 * and return the record (or rather, the ADGameDescription
 	 * contained in it) for which all files described by it are present, and
 	 * among those the one with the maximal number of matching files.
+	 *
 	 * In case of a tie, the entry coming first in the list is chosen.
 	 *
-	 * @param allFiles	a map describing all present files
-	 * @param fslist	a list of nodes for all present files
-	 * @param fileBasedFallback	a list of ADFileBasedFallback records, zero-terminated
-	 * @param filesProps	if not 0, return a map of properties for all detected files here
+	 * @param allFiles           Map describing all present files
+	 * @param fileBasedFallback  List of ADFileBasedFallback records, zero-terminated.
 	 */
 	ADDetectedGame detectGameFilebased(const FileMap &allFiles, const ADFileBasedFallback *fileBasedFallback) const;
 
 	/**
-	 * Compose a hashmap of all files in fslist.
-	 * Includes nifty stuff like removing trailing dots and ignoring case.
+	 * Compose a hashmap of all files in @p fslist.
+	 *
+	 * Removes trailing dots and ignores case in the process.
 	 */
 	void composeFileHashMap(FileMap &allFiles, const Common::FSList &fslist, int depth, const Common::String &parentName = Common::String()) const;
 
 	/** Get the properties (size and MD5) of this file. */
 	bool getFileProperties(const FileMap &allFiles, const ADGameDescription &game, const Common::String fname, FileProperties &fileProps) const;
 
-	/** Convert an AD game description into the shared game description format */
+	/** Convert an AD game description into the shared game description format. */
 	virtual DetectedGame toDetectedGame(const ADDetectedGame &adGame) const;
 
-	friend class FileMapArchive; // for FileMap
+	/** @todo Doc required */
```

Comment:
As discussed in the other PR, I don't think there is a need to document friends.